### PR TITLE
update libraries to improve compatibility with newer versions of react

### PR DESCRIPTION
**Pull request by mt-rsobachi**
## Summary

Updates `react` and all other libraries to improve compatibility with newer versions of our apps.

## Solution

- Upgraded `react` and other libraries
- Modified script to support new libraries

## Test

- [x] Test with Universal Vault (react `^16.12.0`)
- [x] Test with Moneytree Business (react `^18.2.0`)
- [x] Test with LINK Kit (react `^17.0.2`)

### CODE REVIEW

**Comment by mt-dfrey** [comment]
the SVG files all look the same, but the PR shows them as changed - is that intentional?

**Comment by mt-rsobachi** [comment]
@mt-dfrey SVGs are created automatically by a `pre-commit` command. I can spot a few very small differences between the old and new SVGs.

File: README.md [hunk 1]
```diff
@@ -48,44 +45,44 @@ yarn build
 ### Icon component:
 
 ```javascript
-import React, { ReactElement } from 'react';
-import Icon from 'mt-web-icons';
+import React from "react";
+import Icon from "mt-web-icons";
 
-export default function IconAuto(): ReactElement {
+export default function IconAuto(): JSX.Element {
```
Comment (mt-max) [inline README.md:51]:
```suggestion
export default function IconAuto(): ReactNode {
```
does this work? It is a collection of the supported types

File: README.md [hunk 2]
```diff
@@ -48,44 +45,44 @@ yarn build
 ### Icon component:
 
 ```javascript
-import React, { ReactElement } from 'react';
-import Icon from 'mt-web-icons';
+import React from "react";
```
Comment (mt-max) [inline README.md:48]: what's up with the colon change?

Comment (mt-rsobachi) [reply README.md]: prettier

**Review by mt-max — APPROVED** [review]

**Review by mt-max — APPROVED** [review]

**Pull request merged**